src/Compilers/Server/VBCSCompiler/CompilerRequestHandler.cs

```diff
@@ -41,7 +41,8 @@ public RunRequest(Guid requestId, string language, string? workingDirectory, str
 
     internal sealed class CompilerServerHost : ICompilerServerHost
     {
-        public IAnalyzerAssemblyLoader AnalyzerAssemblyLoader { get; } = new ShadowCopyAnalyzerAssemblyLoader(baseDirectory: Path.Combine(Path.GetTempPath(), "VBCSCompiler", "AnalyzerAssemblyLoader"));
+        public IAnalyzerAssemblyLoader AnalyzerAssemblyLoader { get; } = new ShadowCopyAnalyzerAssemblyLoader(
+            baseDirectory: Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "VBCSCompiler"));
```

Comment:
The behavior on Windows today is current as the temp directory is per user. There is already a lot of expectation around that being the location of shadow copies and I do not want to change that. Think this change needs to be limited to non-Windows platforms. 

Further I think the better strategy on non-Windows is keeping it in `/tmp` but appending user specific folder to it. Likely just the user name or id.

Reply:
Do you have something like `/tmp/VBCSCompiler/user` in mind? This would not work, because `/tmp/VBCSCompiler` would be owned by the first user. What do you think about using the previous approach (`/tmp/VBCSCompiler-user`) for non-Windows only?